ouroboros-network/src/Ouroboros/Network/NodeToNode.hs

```diff
@@ -223,16 +230,23 @@ connectTo_V1 tracers versionData application localAddr remoteAddr =
 -- * when an async exception is thrown to kill the main thread the server thread
 --   will be cancelled as well (by 'withAsync')
 --
+-- TODO: specialise to use `Socket`, node-to-node protocol is only used over
+-- TCP sockets.
```

Comment:
I don't think we should limit node-to-node to only TCP.

Reply:
I agree; I'll update this comment.

Reply:
Fixed; I finished the un-finished todos.  `NodeToNode` module is using `SocketSnocket` everywhere.  It's a public interface, so we don't want to be more general (polymorphic) than it is needed.